### ISSUE TITLE
Panning with chart.plotOptions causes zooming issue.

### ISSUE THREAD

**Issue by JoelDorrington** [highcharts/highcharts#9612]
#### Expected behaviour
Configure highcharts to zoom and pan:
`chart: {
  zoomtype: 'x',
  panning: true,
  panKey: 'ctrl'
},
plotOptions: {
  series: {
     pointPlacement: 'on'
   }
}`
When holding the ctrl button chart should pan smoothly and remain at the same zoom level.

#### Actual behaviour
When holding ctrl and dragging the mouse, the chart pans but also zooms, slowly if zoom is low, but gets exponentially more pronounced as the zoom increases.

The plotOptions property seems to be the culprit upon first inspection. Removing the option produces expected behaviour.

#### Live demo with steps to reproduce

Configure highcharts with the above options.
I'm unsure if any other options affect the issue.

Here is partial a reproduction of my chart.
http://jsfiddle.net/wmqfryh8/3/

#### Product version

6.2.0
#### Affected browser(s)
Latest Chrome and Firefox, maybe others I didn't test.

**Comment by pawelfus** [comment]
Hi @JoelDorrington 

Thank you for reporting the issue! You are right, it's broken.

**Simplified demo:**
http://jsfiddle.net/BlackLabel/s8eqzg27/

In short data distribution and `pointPlacement = "on"` causes the issue. When using default value ("between") it works fine: http://jsfiddle.net/BlackLabel/s8eqzg27/2/

**Comment by sebastianbochan** [comment]
Unstale

**Comment by michahagg** [comment]
Hi @JoelDorrington and @pawelfus,

I have the same issue with my column-chart: https://jsfiddle.net/m4xbowr7/ 

When you zoom in and with the ALT-key try to pan the chart, the chart goes everywhere and then refuses to pan all together.

Remarkable is that when I change 'between' to 0.5, which induces almost the same behavior, panning does work as expected. It's just that with 0.5, the columns at left or right edge or both are drawn differently. That is not what I want.

I did take a quick look at the code. Might it have something to do with 'between' and 'on' not being a number, where 0.5 is?

With kind regards,
Micha

**Comment by pawelfus** [comment]
Thank you for the details @michahagg - that's correct, `between`/`on` option works a bit different than numeric value. Numeric value does increase the extremes. That extra space at the edges are the reason why it's broken.

**Simplified demo:**
https://jsfiddle.net/BlackLabel/6z1tkan3/1/ 
Steps: drag&drop to pan.

Alternative workaround is to set `series.pointRange = 0` - however that will render chart almost the same as with numeric value for `pointPlacement`. We can improve this by setting `minPadding` and `maxPadding`: https://jsfiddle.net/BlackLabel/091kpg5q/1/ but when zoomed in, first and last column will be cut off.

*Internal note:*
It seems that `halfPointRange` is used twice: in `Axis.toValue()` and in `pan()` method it's added/subtracted again. Probably it shouldn't happen when `pointPlacement` option is set as a string value?

**Comment by michahagg** [comment]
Hi @pawelfus, I have found the fix for the highcharts column datetime x-panning scenario. I made the fix in the highcharts.src.js file (v8.2.0). And then transferred those changes to the corresponding ts-files in the Highcharts repo. From those I made a patch file that I attached to this comment.

I have **only** tested it in the situations where I have a regular highchart, with a column and a spline series. And with the values 'on', 'between', 0, 0.5 and -0.5 for 'chart.plotOptions.column.pointPlacement'.

I had forked and cloned the highcharts GitHub repo, but couldn't get it setup as I don't have Visual Studio. Unfortunately I couldn't run the tests and make this into a proper PR. Could you do this?

Thanks in advance!

[Fix_panning_with_pointPlacement_(#9612).zip](https://github.com/highcharts/highcharts/files/5468273/Fix_panning_with_pointPlacement_.9612.zip)

**Comment by pawelfus** [comment]
Thank you for sharing your fix, tested it. Unfortunately, this breaks basic panning in charts with simple line series (for example this one: https://jsfiddle.net/BlackLabel/qg36tpce/1/ zoom-in, then try panning). I think removing `minPixelPadding` from `Axis.translate` is quite risky. But if this works for you then it's okay

**Comment by michahagg** [comment]
Hi @pawelfus, I was afraid that I broke something with my rigorous change. It does work for me in my case (and my case only I guess.)

I was hoping that you or someone else at Highcharts would take my fix and use it to create a definitive fix for the issue. Is there any chance it is going to be picked up soon?

**Comment by pawelfus** [comment]
Thank you for trying! Really appreciate this

We will investigate the issue and let you know the details soon

**Comment by pawelfus** [comment]
*Internal note:*
When fixing, focus on `pan()` method as only panning cause the issue. There are no premises we should make any changes in `axis.translate()` method. Probably just checking in `pan()` for the `pointPlacement` options is a good start.

**Comment by michahagg** [comment]
Hi @pawelfus I have a more detailed issue report. I made a JSFiddle that allows you to see the fix working in a life example. It also shows 'programmatically' how the fix for pointPlacement='on' differs from the fix for 'between', -0.5, 0 and 0.5.

https://jsfiddle.net/dv5tj138/

In the HTML pane it loads Highcharts, but also overrides the functions Axis.prototype.translate and Chart.prototype.pan. The code that needs changed is replaced with a function that is defined at the top of the HTML pane. This is where the changed code is isolated from the rest.
The const POINTPLACEMENT can be set to 'between', 'on', -0.5, 0 and 0.5 to play with the panning in the different situations. And to activate the proper fix for that situation as well.

Hope this helps you finding a solution.

**Comment by pawelfus** [comment]
Thank you for your help @michahagg ! I really appreciate this

**Comment by michahagg** [comment]
Hello @pawelfus, it's been some time since I looked into this. In the mean time, I've updated Highcharts in my application to version 9.1.0. The weird panning behavior is still present. I was hoping a definitive solution would have been found and implemented. I'm not happy having to patch the highcharts library each time I update to the latest version.

Is there something I can do to help find the proper fix?

Thanks in advance.
